File: algorithm/BOJ/1715/Main.cpp
#include <cstdio>
#include <queue>
/**
 * https://www.acmicpc.net/problem/1715
 * BOJ 백준온라인져지 1715 카드 정렬하기 풀이
 */
using namespace std;
int main () {
    int cardCount;
    scanf("%d", &cardCount);
    priority_queue<int, vector<int>, greater<int> > queue;
    while (cardCount--) {
        int temp;
        scanf("%d", &temp);
        queue.push(temp);
    }
    int result = 0;
    while (queue.size() > 1) {
        // a + b + ((a + b) + c)
        // result = (a + b)
        int a = queue.top(); queue.pop();
        int b = queue.top(); queue.pop();
        result += a + b;
        queue.push(a + b);
    }
    printf("%d", result);
}

File: algorithm/BOJ/1715/Main.java
import java.util.*;
import java.io.*;

/**
 * https://www.acmicpc.net/problem/1715
 * BOJ 백준온라인져지 1715 카드 정렬하기 풀이
 */
public class Main {
    private static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int sum = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i < N; i++) {
            pq.offer(Integer.parseInt(br.readLine()));
        }
        while (pq.size() >= 2) {
            int temp = pq.poll() + pq.poll();
            pq.offer(temp);
            sum += temp;
        }
        bw.write(String.valueOf(sum));
        bw.flush();
    }
}